### PR TITLE
Setup Karma  for unit testing

### LINKED ISSUE #448: Add Karma

### PR DESCRIPTION
**Pull request by jleft**
Resolves #448

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -454,7 +442,7 @@ module.exports = function(grunt) {
     grunt.registerTask('deploy', ['buildAndTest', 'gh-pages:origin']);
     grunt.registerTask('deploy:upstream', ['buildAndTest', 'gh-pages:upstream']);
 
-    grunt.registerTask('buildAndTest', ['build', 'test']);
+    grunt.registerTask('buildAndTest', ['build', 'test:phantom']);
     grunt.registerTask('dev', ['connect:watch', 'watch']);
```
Comment (tlsim) [inline Gruntfile.js]: Might be nice to have `grunt dev` start a Karma server (Chrome only?) with `grunt watch` triggering running tests. https://github.com/karma-runner/grunt-karma#karma-server-with-grunt-watch

Comment (jleft) [inline Gruntfile.js]: I'm not sure about this, at least with the current set of tests. The tests only cover a few utilities, so unless we're changing these utilities, then it's unlikely that the tests need to be run so frequently when developing. The `grunt test:chrome` task will watch for any changes and re-run the tests. Happy to combine the two into another task, if that'd be useful?

Comment (jleft) [inline Gruntfile.js]: I've added the task `devTest` - let me know what you think :)

Comment (tlsim) [inline Gruntfile.js]: :+1:

**Pull request merged**